spec/ScheduledPushRouter.spec.js

```diff
@@ -0,0 +1,355 @@
+'use strict';
+
+const request = require('../lib/request');
+
+describe('Scheduled Push Router', () => {
+  // the number of milliseconds to wait for the push to save before continueing
+  // This MAY need to be increase if tests fail on a slower box
+  // 5 ms results in 1/3 failing
+  // 10 ms results in all passing on a fast box
+  // 50 ms is a 5x buffer for a slower box
+  const delayToSave = 50;
+
+  const delayPromise = delay => {
+    return new Promise(resolve => {
+      setTimeout(resolve, delay);
+    });
+  };
+
+  const setup = async () => {
+    // const sendToInstallationSpy = jasmine.createSpy();
+
+    const pushAdapter = {
+      send: function() {
+        return Promise.resolve({
+          err: null,
+          transmitted: true,
+        });
+      },
+      getValidPushTypes: function() {
+        return ['ios', 'android'];
+      },
+    };
+
+    await reconfigureServer({
+      scheduledPush: true,
+      appId: Parse.applicationId,
```

Comment:
do you need to reset masterkey, appid, serverurl?  isn't it already set if you don't do?